main/java/mjc/beans/AbstractPurchase.java

```diff
@@ -0,0 +1,76 @@
+package main.java.mjc.beans;
+
+import main.java.mjc.exceptions.CsvLineException;
+
+public abstract class AbstractPurchase implements Comparable<AbstractPurchase>, Cloneable {
```

Comment:
```suggestion
public abstract class AbstractPurchase implements Comparable<AbstractPurchase> {
```
[Avoid clone](http://www.javapractices.com/topic/TopicAction.do?Id=71). 
J. Bloch says the same: Override clone judiciously. 3rd edition, item 13.